templates/dataset_inplace.cpp.in

```diff
@@ -26,7 +26,7 @@ DataArray @NAME@(DataArray &&lhs, const Variable &rhs) {
 DataArray @NAME@(DataArray &&lhs, const DataArray &rhs) {
   expect::coordsAreSuperset(lhs, rhs);
   union_or_in_place(lhs.masks(), rhs.masks());
-  return @NAME@(lhs, rhs.data());
+  return std::move(@NAME@(lhs, rhs.data()));
```

Comment:
Why is the move needed here?

Reply:
I found it after deleting the copy constructor of DataArray. But the `move` was actually in the wrong place, see the new version.
`@NAME@(lhs, rhs.data())` called the lvalue reference overload which returns an lvalue ref and therefore requires a copy here.